php/class-post-type.php

```diff
@@ -1331,4 +1356,91 @@ public function handle_snapshot_fork() {
 		$forked_post['edit_link'] = get_edit_post_link( $forked_post_id, 'raw' );
 		wp_send_json_success( $forked_post );
 	}
+
+	/**
+	 * Filter snapshot split data and create another snapshot.
+	 *
+	 * @param string $content post_content of post.
+	 *
+	 * @return string return post_content.
+	 */
+	public function filter_snapshot_split_data( $content ) {
+		global $post;
+		$post_type_object = get_post_type_object( static::SLUG );
+		$split_key = 'split-snapshot';
+
+		$should_filter_content = (
+			isset( $post->post_status )
+			&&
+			( 'publish' !== $post->post_status )
+			&&
+			current_user_can( $post_type_object->cap->edit_post, $post->ID )
+			&&
+			( static::SLUG === $post->post_type )
+			&&
+			! empty( $_REQUEST[ $split_key ] )
+			&&
+			is_array( $_REQUEST[ $split_key ] )
+			&&
+			isset( $_REQUEST[ static::SLUG ] )
+			&&
+			wp_verify_nonce( $_REQUEST[ static::SLUG ], static::SLUG . '_settings' )
+			&&
+			! ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE )
+		);
+
+		if ( ! $should_filter_content ) {
+			return $content;
+		}
+		$setting_ids_to_split = $_REQUEST[ $split_key ];
+		$data = json_decode( wp_unslash( $content ), true );
+		$new_split_data = array();
+
+		foreach ( $setting_ids_to_split as $i => $key ) {
+			if ( isset( $data[ $key ] ) ) {
+				$new_split_data[ $key ] = $data[ $key ];
+				unset( $data[ $key ] );
+			}
+		}
+		if ( empty( $new_split_data ) ) {
+			return $content;
+		}
+
+		$uuid = wp_generate_uuid4();
+		$new_post_arr = array(
+			'post_type' => static::SLUG,
+			'post_content' => Customize_Snapshot_Manager::encode_json( $new_split_data ),
+			'post_status' => 'draft',
+			'post_author' => get_current_user_id(),
+			'post_name' => $uuid,
+			'post_title' => $uuid,
+		);
+		$theme = get_post_meta( $post->ID, '_snapshot_theme', true );
+		if ( ! empty( $theme ) ) {
+			$new_post_arr['meta_input']['_snapshot_theme'] = $theme;
```

Comment:
Should we include meta about the source changeset that it came from? Wouldn't this be like a fork?

Reply:
We can add it, But where would we use it? If we are showing source from where changeset was split from that would work. Should i add that? @westonruter